### PR TITLE
[feat] style log in page

### PR DESCRIPTION
**Pull request by ccatherinetan**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* styled the log in page
* new login page route: `src/app/auth/login` (changed the route for sign out and reset password) 
* added styled component H1 (for "Log In" page title), H3, and H3Centered (used for "Forgot your password?" in `styles/text`
* removed Sign-Up functionality (to be implemented in the next sprint)

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."
Screen Shot 2023-11-02 at 4 15 18 PM

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
Check TextInput: `components/TextInput/index` for updated TextInput props; `components/TextInput/styles` for updated font (Inter → Open Sans) 
Check `src/app/auth/login` for styling 

Log in: 
Go to http://localhost:3000/auth/login, enter a valid email + password (check supabase for latest sign-in time) 
If successfully signed in, you should be rerouted to the main page. 
Test that reset-password and sign-out reroute to the new login page. 
Sign Out: http://localhost:3000/settings → click "Sign Out" 
Reset password: http://localhost:3000/settings → click "Reset Password" (or go to  http://localhost:3000/reset-password, or click "Forgot your password?") 
* "Forgot your password?" currently leads to Reset Password, but we'll change that later

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

* error handling of email and password inputs 
* styling the sign in button + sign up link
* Styling the hyperlinks Forgot Password + Sign In → **EDIT: FIXED!!!** 

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: src/app/auth/login/page.tsx
Comment (jinkang-0) [inline src/app/auth/login/page.tsx:5]: no need to do `/index`, leaving it as `@/components/TextInput` will do the same (`index.js/jsx` should be the default that Typescript searches for in the import path)

File: src/app/auth/login/page.tsx
Comment (jinkang-0) [inline src/app/auth/login/page.tsx]: After some thought, maybe it'll be cleaner to discard the `SixteenDiv`? Then add margins to `ForgotPassword` and `H4Centered` (or another component) instead, which would reduce the element nesting hell rn

File: src/app/auth/login/page.tsx
Comment (jinkang-0) [inline src/app/auth/login/page.tsx]: 😆 maybe put "Password" instead

File: src/app/input-text/page.tsx
Comment (jinkang-0) [inline src/app/input-text/page.tsx]: should we remove this file? since it's only for testing

File: src/components/TextInput/index.tsx
Comment (jinkang-0) [inline src/components/TextInput/index.tsx:37]: I think we can take advantage of the fact that `errorText` can be an empty string and use that to check if it's erroring or not, which could save us space by not taking in `erroring`

File: src/styles/text.tsx
Comment (jinkang-0) [inline src/styles/text.tsx]: do we need this anywhere...? (also `color: inherit` is the default, I don't think this does anything)

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: should we use `@/` here as well?

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: if it is normal, do we need to include this line?

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: is this for adding space between the text and the rest of the input? if so, i would recommend using `padding` instead with `rem` units, I don't particularly trust `line-height` as a CSS property... (feel free to prove me wrong)

File: src/app/auth/styles.tsx
Comment (jinkang-0) [inline src/app/auth/styles.tsx]: do we need this? I'm almost certain `p` tags have normal `font-style` by default

File: src/app/auth/styles.tsx
Comment (jinkang-0) [inline src/app/auth/styles.tsx]: for a significant amount in `px` (i.e. somewhere in the ballpark of >5), we should probably consider using `rem` instead? although there have been cases where they don't look the same (despite having the same value after unit conversion), so feel free to disregard this if that is the case

File: src/app/auth/login/page.tsx
Comment (jinkang-0) [inline src/app/auth/login/page.tsx]: should we use `@/` here?

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: as far as i recall, `input` can't have children elements? (they're usually all self-closing)
unsure what the `gap` would have an effect on here

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: could consider tossing this into the default (not `:focus`) (without `!important`) (usually all that's needed) and see if it still works, because ideally we want to avoid `!important` if we can help it

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: thoughts on clearing out this comment if we won't use `H4` in here?

**Review by jinkang-0 — CHANGES_REQUESTED** [review]
some minor stuff but looks good!

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx:5]: gotcha tysm!

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx]: mmmm, the problem is that the gap in the main div is 40 px, so I would still need to make a div for the 2 questions + the Forgot Password. I think it's best to keep the div?

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx]: @davidqing6432 Thoughts on placeholder text for consistency? I might change it to "Password" instead of "Input Suggestion" for this field. But do we have/need a general  rule for placeholder text?

File: src/app/input-text/page.tsx
Comment (ccatherinetan) [reply src/app/input-text/page.tsx]: Rahi is still using it for her testing, so Pragya said to keep this file for now.

File: src/styles/text.tsx
Comment (ccatherinetan) [reply src/styles/text.tsx]: I'll prob delete this, since i was just testing it out and it didn't work. thanks for catching it!

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: yes! done

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: noted! i deleted it.

File: src/app/input-text/page.tsx
Comment (rahihazra) [reply src/app/input-text/page.tsx]: I created my own file for testing, so you can go ahead and delete it!

File: src/styles/text.tsx
Comment (rahihazra) [inline src/styles/text.tsx:51]: great work overall!!!!!

File: src/app/auth/styles.tsx
Comment (ccatherinetan) [reply src/app/auth/styles.tsx]: gotcha! i deleted the normal

File: src/app/auth/styles.tsx
Comment (ccatherinetan) [reply src/app/auth/styles.tsx]: gotcha! i think it looks good when i deleted it!

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx]: fixed!

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: good catch! i deleted the gap.

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: yes! i'll delete!

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: deleted the line-height! i think it was carried over from the old designs.

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: gotcha!

File: src/components/TextInput/index.tsx
Comment (ccatherinetan) [reply src/components/TextInput/index.tsx:37]: ok! i think a future sprint will handle error handling; i might just leave it as is right now.

File: src/app/auth/login/page.tsx
Comment (jinkang-0) [reply src/app/auth/login/page.tsx]: ahh I see, perhaps instead of using gap in the big div, we add vertical margin of 40px to the `QuestionDiv`? This is really minor and only saves us one `div` nesting, so feel free to decide whether or not to implement that!

**Review by jinkang-0 — APPROVED** [review]

File: src/app/auth/styles.tsx
Comment (pragyakallanagoudar) [inline src/app/auth/styles.tsx:13]: Clarifying question: Is this pulled from Alvaro's interest form PR?

File: src/app/auth/login/page.tsx
Comment (pragyakallanagoudar) [inline src/app/auth/login/page.tsx]: Screenshot 2023-10-28 at 9 52 26 PM

We may want to handle this in a future PR (possibly the `layout.tsx` one) but wondering if it would be possible to resize and center the blue border div so the screen looks like the one in the design?

When I initially wrote this PR I only added the header/questions/button on the Linear issue, with the assumption that the blue div & positioning would be handled in a future sprint. However since we already have the blue div here (and it looks great!) perhaps we can just work on positioning right now.

File: src/styles/text.tsx
Comment (pragyakallanagoudar) [inline src/styles/text.tsx:48]: big ++ for adding text styles!!

File: src/app/layout.tsx
Comment (pragyakallanagoudar) [inline src/app/layout.tsx]: YAY thank you for doing this

File: src/app/auth/styles.tsx
Comment (ccatherinetan) [reply src/app/auth/styles.tsx:13]: I'm assuming you're referring to https://github.com/calblueprint/immigration-justice-project/pull/25. If so, no, I created my own styled div and didn't intentionally pull this from Alvaro's PR (am i supposed to?). sorry if the component name is a bit confusing; let me know if there's a better naming convention to follow, or I'll try to think of a better name.

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx]: sounds good! I didn't know if that was supposed to be handled in the layout, or in the div. Let me actually do my reading on layouts 😅 to see if my statement makes any sense. but otherwise, yes, i'll def include the div positioning in this sprint.

File: src/app/auth/styles.tsx
Comment (pragyakallanagoudar) [reply src/app/auth/styles.tsx:13]: Oh no worries! I was just curious because of the name and the fact that there's a similar div in his PR (I believe he calls it `CaseInterestContainer`.) This is totally fine though, we can potentially consider condensing them into one component once both PRs are merged in.

File: src/app/auth/login/page.tsx
Comment (surajr10) [reply src/app/auth/login/page.tsx]: i think a standardized div to accomplish the blue box should be okay and we can avoid using a layout file. usually we can keep the layout for a consistent component that we'll see across multiple pages (ex. navbars, headers, etc.)

File: src/app/auth/styles.tsx
Comment (surajr10) [inline src/app/auth/styles.tsx:16]: nit: can probably remove the px value comments and put a conversion factor that you used at like the top of the file as a comment, or if you want you could set a variable for the conversion factor and actually calculate out the rem values for each css field which might be helpful if all the figma stuff is in pixels, but that might be a lil tedious, so totally up to you!

File: src/app/auth/styles.tsx
Comment (surajr10) [inline src/app/auth/styles.tsx:1]: all the styled-components style files can have the .ts extension instead of .tsx b/c they aren't returning out jsx. mostly just housekeeping.

File: src/components/TextInput/styles.tsx
Comment (surajr10) [inline src/components/TextInput/styles.tsx:12]: noticed that the border weight here is 1 px on figma when i inspected with dev move (can check w/ david what he thinks looks appropriate). didn't check other specific values, but could be good to do a quick check that the styles match figma

File: src/app/auth/login/page.tsx
Comment (surajr10) [reply src/app/auth/login/page.tsx]: also in a similar vein for this task, this might involve starting to use media queries but not sure if this is exactly necessary and i also think that building out responsiveness can be a later task. @pragyakallanagoudar thoughts?

[pragya] for some reason i can't respond to this but i can edit it ???? anyways this is a great point, and i think we can come back to responsiveness later so Catherine no need to worry about this rn!

File: src/styles/colors.ts
Comment (surajr10) [inline src/styles/colors.ts:1]: a global colors file is great! i might recommend using ThemeProvider alongside this so that you don't have to import these colors in sm parts of the app. this could potentially be done in a layout file in the root of the app directory

File: src/styles/text.tsx
Comment (surajr10) [inline src/styles/text.tsx]: i think we can take out specific text styles like h4centered and keep more consistent general ones like just h4. instead, we can style children tags that live inside a container. for example inside the FormDiv you have, you can put this (H4 {
    text-align: center;
  }) which would achieve the same behavior

**Review by surajr10 — COMMENTED** [review]
overall wow great PR! this is some great stying and use of global colors/fonts, organized style components, etc.! lmk if you have any questions about anything i commented

File: src/components/TextInput/index.tsx
Comment (surajr10) [inline src/components/TextInput/index.tsx]: think typing this as a regular function should be fine for the most part but i'd recommend typing it as Dispatch<SetStateAction<string>> as that's the type of the setState function that is returned from useState

File: src/components/TextInput/styles.tsx
Comment (surajr10) [inline src/components/TextInput/styles.tsx]: yee it is! you can make this InputText = styled(H4) and then in the actual jsx portion you can pass in the prop "as='input'" to make sure that the component is still being rendered as an input component

File: src/app/auth/login/page.tsx
Comment (surajr10) [inline src/app/auth/login/page.tsx]: nit: prob rename from sixteen to like spacer div just so it's a lil bit easier to understand when you first see it

File: src/app/auth/login/page.tsx
Comment (surajr10) [reply src/app/auth/login/page.tsx]: yea i agree, might be a lil cleaner without the `SixteenDiv` but i kinda like the use of gap over margin here so up to you! as jinkang said, it's pretty minor

File: src/app/auth/styles.tsx
Comment (ccatherinetan) [reply src/app/auth/styles.tsx:16]: gotcha, i might  keep it as is for this PR, but i'll look out for this in future PRs!

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx:12]: gotcha! yeah david said he would change border thickness to 2px on figma later. but the border-radius refers to radius of the curvature of the border! thanks for the reminder!

File: src/components/TextInput/styles.tsx
Comment (ccatherinetan) [reply src/components/TextInput/styles.tsx]: gotcha, thank you! this is super helpful, suraj!

File: src/app/auth/login/page.tsx
Comment (ccatherinetan) [reply src/app/auth/login/page.tsx]: gotcha! i've renamed it!

File: src/styles/colors.ts
Comment (ccatherinetan) [reply src/styles/colors.ts:1]: gotcha! i'll defer to @pragyakallanagoudar about using ThemeProvider!

File: src/app/auth/styles.tsx
Comment (ccatherinetan) [reply src/app/auth/styles.tsx:1]: gotcha! changed!

File: src/components/TextInput/index.tsx
Comment (ccatherinetan) [reply src/components/TextInput/index.tsx]: Consulting @pragyakallanagoudar about this!

File: src/styles/colors.ts
Comment (pragyakallanagoudar) [reply src/styles/colors.ts:1]: will look into this, but you don't have to worry about this rn!

File: src/components/TextInput/index.tsx
Comment (ccatherinetan) [reply src/components/TextInput/index.tsx]: implemented!

**Review by pragyakallanagoudar — APPROVED** [review]
LOOKS FANTASTICCCCCCCCCC thank you for all your hard work on this 🤩🤩🤩🤩

**Pull request merged**
